source/Mlos.Python/mlos/Optimizers/unit_tests/TestBayesianOptimizer.py

```diff
@@ -353,3 +369,40 @@ def test_bayesian_optimizer_default_copies_parameters(self):
         assert original_config.min_samples_required_for_guided_design_of_experiments == 10
         print(original_config.experiment_designer_config.fraction_random_suggestions)
         assert original_config.experiment_designer_config.fraction_random_suggestions == .5
+
+    def validate_optima(self, optimizer: OptimizerBase):

```

Comment:
Do you want to put this in a shared base class so as not to copy &paste? Or are there changes?

Reply:
No changes, that's a good comment in general - there is a bunch of repeated code in our tests.

Reply:
Since we are using the self.assertRaises, the validate_optima() method is actually bound to a specific unittest.TestCase instance. Sharing this code would require a shared parent in the inheritance hierarchy... seems like a lot of work. Instead, I would like to propose:
1. let's move to pytest first - simple asserts will replace the self.assertXYZ methods.
2. turn on linter on unit_tests and clean up everything then - pylint should highlight duplicated code too.